int16 input_para_in_right
int8 input_dir_in_right
int16 input_para_in_left
int8 input_dir_in_left
int32 output_out_data
int32 id
